centreon-certified/opsgenie/opsgenie-event-apiv2.lua

```diff
@@ -0,0 +1,433 @@
+  }
+end
+
+
+function EventQueue:format_event_ba()
+  local category = self.sc_event.event.category
+  local element = self.sc_event.event.element
+  self.sc_event.event.formated_event = {
+    message= tostring(self.sc_event.event.cache.ba_name) .. " is " .. self.sc_param.params.status_mapping[category][element][self.sc_event.event.state] .. ", health level reached " .. self.sc_event.event.level_nominal,
+    tags = {}
+  }
+
+  --Handle tags
+  if self.sc_event.event.cache.ba_name and self.sc_event.event.cache.ba_name ~= nil then
+    if self.sc_param.params.enable_incident_tags == 1 then
+      -- get bvs info (bv_name, bv_description...) if we not have it
+      if not self.sc_event.event.cache.bvs then
+        self.sc_event.event.cache.bvs =self.sc_broker:get_bvs_infos(self.event.host_id)
+      end
+
+      --Insert Tags BV with bvs info
+      for i, bv_data in ipairs(self.sc_event.event.cache.bvs) do
+        table.insert(self.sc_event.event.formated_event.tags, bv_data.bv_name)
+      end
+
+      --Handle custom tags
+      if self.sc_params.params.ba_incident_tags ~= '' then
+        local custom_tags = self.sc_common:split(self.sc_params.params.ba_incident_tags, ',')
+        for i, custom_tag in ipairs(custom_tags) do
+          self.sc_logger:info("[EventQueue:is_valid_bam_event: adding " .. tostring(custom_tag) .. " to the list of tags")
+          table.insert(self.sc_event.event.formated_event.tags, custom_tag)
+        end
+      end
+    end
+  end
+end
+
+--------------------------------------------------------------------------------
+-- EventQueue:build_payload, concatenate data so it is ready to be sent
+-- @param payload {string} json encoded string
+-- @param event {table} the event that is going to be added to the payload
+-- @return payload {string} json encoded string
+--------------------------------------------------------------------------------
+function EventQueue:build_payload(payload, event)
+  if not payload then
+    payload = broker.json_encode(event)
+  else
+    payload = payload .. ',' .. broker.json_encode(event)
+  end
+
+  return payload
+end
+
+--------------------------------------------------------------------------------
+-- EventQueue:send_data, send data to external tool
+-- @return {boolean}
+--------------------------------------------------------------------------------
+function EventQueue:send_data(payload)
+  self.sc_logger:debug("[EventQueue:send_data]: Starting to send data")
+
+  -- write payload in the logfile for test purpose
+  if self.sc_params.params.send_data_test == 1 then
+    self.sc_logger:notice("[send_data]: " .. tostring(payload))
+    return true
+  end
+
+  if self.sc_event.event.categories == 1 then
+    local endpoint = self.sc_params.params.api_url .. url_path
+  else
+    local endpoint = self.sc_params.params.integration_api_token .. url_path
+  end
```

Comment:
there, we have a little big issue :) 

what does it mean ? 
it means that depending on the type of event, we need to send them to a dedicated endpoint.

this is handled by the new lib sc_flush and we have a new parameter called  **send_mixed_events** . 

when do we need it ? when people want to send ba_status events. 

force send_mixed_events to 0 when accepted_categories contains bam

change `EventQueue:send_data(payload)` to  `EventQueue:send_data(payload, info)`  (I'll need to do an update of the sc_flush module)

you'll also need to change  `[1] = function (payload) return self:send_data(payload) end` and add the info parameter

the info method argument will only be available when we don't send mixed events and it will have the following structure 

```lua 
{
  "category_name"= "category_name",
  "category_id" = "category_id",
  "element_name" = "element_name",
  "element_id" = "element_id"
}
```

you'll then be able to transform your condition